ChallengesWithTestsMarkVII/ChallengesSet07.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ChallengesWithTestsMarkVII
+{
+    public class ChallengesSet07
+    {
+        public int CountOfBusinessesWithNegativeNetProfit(List<Business> businesses)
+        {
+            if (businesses == null)
+            {
+                return 0;
+            }
+
+            int count = 0;
+            foreach (Business business in businesses)
+            {
+                if (business.TotalRevenue < business.TotalExpenses)
+                {
+                    count += 1;
+                }
+            }
+            return count;
+        }
+
+        public string GetCommaSeparatedListOfProfitableBusinesses(List<Business> businesses)
+        {
+            if (businesses == null)
+            {
+                return "";
+            }
+
+            string namesOfProfitableBusinesses = "";
+
+            foreach (Business business in businesses)
+            {
+                if (business.TotalRevenue > business.TotalExpenses)
+                {
+                    namesOfProfitableBusinesses += $"{business.Name},";
+                }
+            }
+
+            return namesOfProfitableBusinesses.TrimEnd(',');
+        }
+
+        public string GetNameOfHighestParentCompany(Business business)
+        {
+            string nameOfParentCompany = business.Name;
+
+            while(business.ParentCompany != null)
```

Comment:
Good demonstration of comfort with reference types.  Extra challenge: can you solve this method recursively?